mp/src/game/client/momentum/ui/HUD/hud_stickybombs.cpp

```diff
@@ -0,0 +1,119 @@
+#include "cbase.h"
+#include <vgui/ILocalize.h>
+#include <vgui/ISurface.h>
+#include <vgui/IVGui.h>
+#include <vgui_controls/EditablePanel.h>
+#include <vgui_controls/Label.h>
+#include "c_mom_player.h"
+#include "hud.h"
+#include "hudelement.h"
+#include "iclientmode.h"
+#include "ienginevgui.h"
+#include "mom_system_gamemode.h"
+#include "weapon/weapon_base.h"
+#include "weapon/weapon_mom_stickybomblauncher.h"
+
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+static MAKE_TOGGLE_CONVAR(mom_sj_stickycount_enable, "1", FCVAR_ARCHIVE, "Toggles the stickybomb counter.\n");
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+class CHudStickybombs : public CHudElement, public EditablePanel
+{
+    DECLARE_CLASS_SIMPLE(CHudStickybombs, EditablePanel);
+
+  public:
+    CHudStickybombs(const char *pElementName);
+
+    void ApplySchemeSettings(IScheme *scheme) OVERRIDE;
+    bool ShouldDraw() OVERRIDE;
+    void OnTick() OVERRIDE;
+
+  private:
+    vgui::EditablePanel *m_pStickybombsPresent;
+    vgui::EditablePanel *m_pNoStickybombsPresent;
+    vgui::Label *m_pStickyLabel;
+    vgui::Label *m_pNoStickyLabel;
+};
+
+DECLARE_HUDELEMENT(CHudStickybombs);
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
+CHudStickybombs::CHudStickybombs(const char *pElementName)
+    : CHudElement(pElementName), BaseClass(NULL, "CHudStickybombs")
+{
+    Panel *pParent = g_pClientMode->GetViewport();
+    SetParent(pParent);
+
+    m_pStickybombsPresent = new EditablePanel(this, "StickybombsPresentPanel");
+    m_pNoStickybombsPresent = new EditablePanel(this, "NoStickybombsPresentPanel");
+    m_pStickyLabel = new Label(m_pStickybombsPresent, "NumStickybombsLabel", "");
+    m_pNoStickyLabel = new Label(m_pNoStickybombsPresent, "NumNoStickybombsLabel", "");
+
+    SetHiddenBits(HIDEHUD_MISCSTATUS);
+
+    LoadControlSettings("resource/ui/HudStickybombs.res");
+
+    vgui::ivgui()->AddTickSignal(GetVPanel());
```

Comment:
I'm not a fan of TickSignals as they are called even if the panel isn't showing. Perhaps move the OnTick logic into `OnThink()`, which gets called once per frame (before every `Paint()`), but only if the panel is visible.